src/main/java/it/smartcommunitylab/aac/webauthn/persistence/WebAuthnYubicoCredentialsRepository.java

```diff
@@ -0,0 +1,103 @@
+package it.smartcommunitylab.aac.webauthn.persistence;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.yubico.webauthn.CredentialRepository;
+import com.yubico.webauthn.RegisteredCredential;
+import com.yubico.webauthn.data.ByteArray;
+import com.yubico.webauthn.data.PublicKeyCredentialDescriptor;
+import com.yubico.webauthn.data.PublicKeyCredentialType;
+
+/**
+ * For this class, everytime we should use the 'username', we
+ * will instead use a string in the following format:
+ * 'realmname{@link #separator}username'. This
+ * is due to the fact that yubico's library assumes that two distinct users can
+ * not have the same username (https://git.io/JD5Vr).
+ */
+public class WebAuthnYubicoCredentialsRepository implements CredentialRepository {
+
+    private final String providerId;
+    WebAuthnUserAccountRepository userAccountRepository;
+    WebAuthnCredentialsRepository webAuthnCredentialsRepository;
+
+    public WebAuthnYubicoCredentialsRepository(String provider,
+            WebAuthnUserAccountRepository userAccountRepository,
+            WebAuthnCredentialsRepository webAuthnCredentialsRepository) {
+        this.userAccountRepository = userAccountRepository;
+        this.webAuthnCredentialsRepository = webAuthnCredentialsRepository;
+        this.providerId = provider;
+    }
+
+    @Override
+    public Set<PublicKeyCredentialDescriptor> getCredentialIdsForUsername(String username) {
+        try {
+            WebAuthnUserAccount account = userAccountRepository.findByProviderAndUsername(providerId, username);
+            final Set<WebAuthnCredential> credentials = account.getCredentials();
+            Set<PublicKeyCredentialDescriptor> descriptors = new HashSet<PublicKeyCredentialDescriptor>();
+            for (WebAuthnCredential c : credentials) {
+                PublicKeyCredentialDescriptor descriptor = PublicKeyCredentialDescriptor.builder()
+                        .id(c.getCredentialId()).type(PublicKeyCredentialType.PUBLIC_KEY).transports(c.getTransports())
+                        .build();
+                descriptors.add(descriptor);
+            }
+            return descriptors;
+        } catch (Exception e) {
+        }
+        return new HashSet<>();
```

Comment:
always use collections emptySet instead of creating an empty hashset

Reply:
Fixed in e4e7d2be944dea0af11263e1dfcaa414d22b962b